frontend/src/components/common/StripeCustomer.tsx

```diff
@@ -0,0 +1,118 @@
+import React, { useState } from "react";
+import dotenv from "dotenv";
+import { loadStripe } from "@stripe/stripe-js";
+import {
+  Elements,
+  CardElement,
+  useStripe,
+  useElements,
+} from "@stripe/react-stripe-js";
+
+dotenv.config();
+
+const stripePromise = loadStripe(process.env.REACT_APP_STRIPE_PUBLISHABLE_KEY as string);
+
+const CreateCustomerForm: React.FC = () => {
+  const [name, setName] = useState("");
+  const [email, setEmail] = useState("");
+  const stripe = useStripe();
+  const elements = useElements();
+
+  const handleSubmit = async (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    if (!stripe || !elements) {
+      return;
+    }
+
+    const cardElement = elements.getElement(CardElement);
+
+    if (!cardElement) {
+      console.error("CardElement not found");
+      return;
+    }
+
+    const {
+      error: paymentMethodError,
+      paymentMethod,
+    } = await stripe.createPaymentMethod({
+      type: "card",
+      card: cardElement,
+      billing_details: {
+        name,
+      },
+    });
```

Comment:
It looked like Stripe prevented me from entering my credit card data. Did you encounter anything like this?

![image](https://github.com/user-attachments/assets/ac4eb299-7e56-46d4-912b-c2695c085472)

Reply:
Matt and I reviewed in person - looks like it may have to do with the browser authentication, but you can still create a stripe customer successfully (resolved)